app/views/catalog/_show_main_content.html.erb

```diff
@@ -3,17 +3,21 @@
 <% @page_title = t('blacklight.search.show.title', :document_title => document_show_html_title, :application_name => application_name).html_safe %>
 <% content_for(:head) { render_link_rel_alternates } %>
 
-<div id="document" class="document <%= render_document_class %>" itemscope  itemtype="<%= @document.itemtype %>">
-  <div id="doc_<%= @document.id.to_s.parameterize %>">
-    <%= render_document_partials @document, blacklight_config.view_config(:show).partials %>
-  </div>
-</div>
+<%= render (blacklight_config.show.document_component || Blacklight::DocumentComponent).new(document: @document, component: :div, title_component: :h1, show: true) do |component| %>
+  <% component.with(:footer) do %>
+    <% if @document.respond_to?(:export_as_openurl_ctx_kev) %>
+      <!--
+           // COinS, for Zotero among others.
+           // This document_partial_name(@document) business is not quite right,
+           // but has been there for a while.
+      -->
+      <span class="Z3988" title="<%= @document.export_as_openurl_ctx_kev(@view_context.document_partial_name(@document)) %>"></span>
+    <% end %>
+  <%  end %>
 
-<% if @document.respond_to?(:export_as_openurl_ctx_kev) %>
-  <!--
-       // COinS, for Zotero among others.
-       // This document_partial_name(@document) business is not quite right,
-       // but has been there for a while.
-  -->
-  <span class="Z3988" title="<%= @document.export_as_openurl_ctx_kev(document_partial_name(@document)) %>"></span>
+  <% component.with(blacklight_config.show.document_component.blank? && blacklight_config.view_config(:show).partials.any? ? :body : :partials) do %>
```

Comment:
This is a bit confusing. Why do we need this block if `blacklight_config.view_config(:show).partials` is empty?

Reply:
Someone might have overridden `render_document_partials`?

Reply:
Please add a comment to explain this.  Then it will be easier to refactor once `render_document_partials` becomes a component itself.